jaxrs/src/main/java/com/cerner/beadledom/jaxrs/provider/ForwardedHeaderFilter.java

```diff
@@ -0,0 +1,79 @@
+package com.cerner.beadledom.jaxrs.provider;
+
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.Priority;
+import javax.ws.rs.Priorities;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.PreMatching;
+import javax.ws.rs.ext.Provider;
+
+/**
+ * The ForwardedHeaderFilter reads the request context and determines if the client request was made
+ * from a secure protocol (HTTPS). If the request originated from secure context, it updates the
+ * request to reflect the original protocol.
+ *
+ * <p>To determine if the client request made in a secure context was forwarded by a load balancer
+ * or proxy, the "Forwarded" and "X-Forwarded-Proto" header values are extracted. If either of them
+ * show the request was made from a secure context, the internal request object is updated to reflect
+ * the original secure protocol.</p>
+ *
+ * @author Nick Behrens
+ */
+@Provider
+@Priority(Priorities.HEADER_DECORATOR)
+@PreMatching
+public class ForwardedHeaderFilter implements ContainerRequestFilter {
+  // This regex takes the value of the 'proto=' field and sets it to a group. The 'proto=' field can
+  // be followed by either another key=value pair separated by a ; or it could be the end of the header
+  // Examples:
+  // proto=https => 'protocolValue' group is set to "https"
+  // proto=http;host=localhost => 'protocolValue' group is set to "http"
+  Pattern forwardedPairs = Pattern.compile("proto=(?<protocolValue>[^;]*)(;|\\z)");
```

Comment:
can this be private?

Reply:
yep: https://github.com/cerner/beadledom/pull/149/commits/e28e1da6f2cae0e5ba16371329a4caa0e663684e